### PR TITLE
Tf2

### PR DESCRIPTION
**Pull request by echang264**
Notes on using tf2 in python, as python support is not well documented.

### CODE REVIEW

File: tf2/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Overview
```
Comment (muthuArivoli) [inline tf2/README.md:1]: You should make the overall header as something like #Tf2 Documentation, and these other headers (like overview) to be secondary headers.

File: tf2/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Overview
+
+[tf2](http://wiki.ros.org/tf2) is ROS's newest library for keeping track of robot transforms between coordinate frames. Documentation for tf2 with Python is very limited, so this README is intended to document the use of tf2 transforms for Python 2.7 and provide some helpful resources.
+
+# Simple transform
```
Comment (muthuArivoli) [inline tf2/README.md:5]: Make secondary header

File: tf2/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Overview
+
+[tf2](http://wiki.ros.org/tf2) is ROS's newest library for keeping track of robot transforms between coordinate frames. Documentation for tf2 with Python is very limited, so this README is intended to document the use of tf2 transforms for Python 2.7 and provide some helpful resources.
+
+# Simple transform
+
+The ROS wiki for writing a transform listener has useful info on creating a Buffer and TransformListener: [http://wiki.ros.org/tf2/Tutorials/Writing%20a%20tf2%20listener%20%28Python%29](http://wiki.ros.org/tf2/Tutorials/Writing%20a%20tf2%20listener%20%28Python%29)
```
Comment (muthuArivoli) [inline tf2/README.md:7]: Can you make the hyperlink short? Change the thing in brackets to a word like 'here'.

File: tf2/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Overview
+
+[tf2](http://wiki.ros.org/tf2) is ROS's newest library for keeping track of robot transforms between coordinate frames. Documentation for tf2 with Python is very limited, so this README is intended to document the use of tf2 transforms for Python 2.7 and provide some helpful resources.
+
+# Simple transform
+
+The ROS wiki for writing a transform listener has useful info on creating a Buffer and TransformListener: [http://wiki.ros.org/tf2/Tutorials/Writing%20a%20tf2%20listener%20%28Python%29](http://wiki.ros.org/tf2/Tutorials/Writing%20a%20tf2%20listener%20%28Python%29)
+
+The lookup_transform method is documented for C++ in the following link. The Python method is the same except the name is `lookup_transform`.
+[http://docs.ros.org/indigo/api/tf2_ros/html/c++/classtf2__ros_1_1Buffer.html#acabbd72cae8f49fb3b6ede3be7e34c55](http://docs.ros.org/indigo/api/tf2_ros/html/c++/classtf2__ros_1_1Buffer.html#acabbd72cae8f49fb3b6ede3be7e34c55)
```
Comment (muthuArivoli) [inline tf2/README.md:10]: Shorten hyperlink

File: tf2/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+The lookup_transform method is documented for C++ in the following link. The Python method is the same except the name is `lookup_transform`.
+[http://docs.ros.org/indigo/api/tf2_ros/html/c++/classtf2__ros_1_1Buffer.html#acabbd72cae8f49fb3b6ede3be7e34c55](http://docs.ros.org/indigo/api/tf2_ros/html/c++/classtf2__ros_1_1Buffer.html#acabbd72cae8f49fb3b6ede3be7e34c55)
+
+
+To complete the transform, one can use one of the functions documented in the link below. Although the method names are correct for Python, the methods take `MSGTYPEStamped` rather than simply the `MSGTYPE` (e.g. `PoseStamped` instead of `Pose`). The transform argument should be what is returned from the `lookup_transform` method.
+[http://docs.ros.org/diamondback/api/tf2_geometry_msgs/html/namespacetf2__geometry__msgs_1_1tf2__geometry__msgs.html](http://docs.ros.org/diamondback/api/tf2_geometry_msgs/html/namespacetf2__geometry__msgs_1_1tf2__geometry__msgs.html)
```
Comment (muthuArivoli) [inline tf2/README.md:14]: Shorten hyperlink

File: tf2/README.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+# Simple transform
+
+The ROS wiki for writing a transform listener has useful info on creating a Buffer and TransformListener: [http://wiki.ros.org/tf2/Tutorials/Writing%20a%20tf2%20listener%20%28Python%29](http://wiki.ros.org/tf2/Tutorials/Writing%20a%20tf2%20listener%20%28Python%29)
+
+The lookup_transform method is documented for C++ in the following link. The Python method is the same except the name is `lookup_transform`.
+[http://docs.ros.org/indigo/api/tf2_ros/html/c++/classtf2__ros_1_1Buffer.html#acabbd72cae8f49fb3b6ede3be7e34c55](http://docs.ros.org/indigo/api/tf2_ros/html/c++/classtf2__ros_1_1Buffer.html#acabbd72cae8f49fb3b6ede3be7e34c55)
+
+
+To complete the transform, one can use one of the functions documented in the link below. Although the method names are correct for Python, the methods take `MSGTYPEStamped` rather than simply the `MSGTYPE` (e.g. `PoseStamped` instead of `Pose`). The transform argument should be what is returned from the `lookup_transform` method.
```
Comment (muthuArivoli) [inline tf2/README.md:13]: Use active voice, 'To complete the transform, *you* can use...'

File: tf2/README.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+[http://docs.ros.org/diamondback/api/tf2_geometry_msgs/html/namespacetf2__geometry__msgs_1_1tf2__geometry__msgs.html](http://docs.ros.org/diamondback/api/tf2_geometry_msgs/html/namespacetf2__geometry__msgs_1_1tf2__geometry__msgs.html)
+
+# Example
```
Comment (muthuArivoli) [inline tf2/README.md:16]: Make secondary header

File: tf2/README.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+Below is an example of a simple transform of a PoseStamped message `pose_stamped` from `origin_frame` to `dest_frame`.
+
+```
```
Comment (muthuArivoli) [inline tf2/README.md:20]: Use python highlighting, ```python

File: tf2/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import tf2_ros
+import tf2_geometry_msgs
+
```
Comment (muthuArivoli) [inline tf2/README.md:23]: need to import rospy as well, I think

File: tf2/README.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+tfBuffer = tf2_ros.Buffer()
+listener = tf2_ros.TransformListener(tfBuffer)
+trans = tfBuffer.lookup_transform(dest_frame, origin_frame, rospy.Time(), rospy.Duration(0.5))
+transformed_pose_stamped = tf2_geometry_msgs.do_transform_pose(pose_stamped, trans)
```
Comment (muthuArivoli) [inline tf2/README.md:27]: I think you should make this a function, that takes as parameters the pose_stamped, origin_frame, dest_frame. People will then be able to copy this code and immediately test it out.

File: tf2/README.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
+# Overview
+
+[tf2](http://wiki.ros.org/tf2) is ROS's newest library for keeping track of robot transforms between coordinate frames. Documentation for tf2 with Python is very limited, so this README is intended to document the use of tf2 transforms for Python 2.7 and provide some helpful resources.
```
Comment (muthuArivoli) [inline tf2/README.md:3]: I'm not really understanding what this is contributing - whats wrong with the python tutorials located at http://wiki.ros.org/tf2/Tutorials

**Review by muthuArivoli — CHANGES_REQUESTED** [review]
Alright, I have a few questions/comments.
What is wrong with the python tutorials located at http://wiki.ros.org/tf2/Tutorials?

Make a new header, and make all of your existing headers as secondary headers.

Make all of the hyperlinks short, so that people looking at the docs won't see the links - they will click on a word or phrase.

Comments below go into more detail. Let me know about your answer for the first question.

File: tf2/README.md (hunk 6 above)
Comment (rabinowitz) [reply tf2/README.md:13]: Or the imperative: 'To complete the transform, use...'

**Review by rabinowitz — COMMENTED** [review]
I pretty much agree with all of Muthu's comments and will check back once they're addressed.

**Comment by rabinowitz** [comment]
Can you also add a bullet point to the main repo readme?

**Comment by echang264** [comment]
Comments addressed. Purpose is providing clarification on ```do_transform_____``` functions as the documentation is missing/incorrect.

**Review by muthuArivoli — APPROVED** [review]
LGTM!

**Review by rabinowitz — APPROVED** [review]

**Pull request merged**